### ISSUE TITLE
Deleting a rule does not stop alerts

### ISSUE THREAD

**Issue by johnsusek** [Yelp/elastalert#2043]
If you have a problematic rule sending a lot of alerting, once you delete it the alerts continue arriving. This is because the elastalert_status documents in ES are still there. Is this by design, or a bug? I can try to put in a PR for this, to delete the elastalert_status documents when a rule file is removed while the daemon is running, if this is just not implemented yet.

**Comment by Qmando** [comment]
If you delete the file, it should stop running that rule. There's some caveats here:
- It may seem like it takes two cycles because it checks for rule changes at the end, not beginning
- It won't stop running it if you have `--pin-rules` argument
- If elastalert is spamming alerts because it got 100000 matches and you delete the rule, it's going to work that that list first. You'd need to stop the elastalert process in this case

**Comment by dmt-xo** [comment]
Hi, 
I am running elastalert as a kubernetes deployment in my system. It is working well with CREATE and UPDATE operations, but seems to be failing on DELETE OPERATION, the problem being, once I delete the rule, the logs show "Rule file test.yaml not found, stopping rule execution" but the elasticsearch logstash index still keeps on being queried as per the rule and that too with the alerts being posted even after the deletion of the rule. 
I saw that you have added a provision of is_enabled: False as a way to disable rules but that too apparently doesn't work here, saying: "Rule file check.yaml is now disabled" but the querying still goes on and the alerts are posted. 

Any suggestions or workarounds?

**Comment by tokarthik** [comment]
any workaround I am also looking for the fix

**Comment by Shashankft9** [comment]
@tokarthik 
clone the master branch instead of using the released version, it has the solution. ref: https://github.com/Yelp/elastalert/pull/2484

**Comment by jonscyr** [comment]
https://github.com/Yelp/elastalert/pull/2826

**Comment by Shashankft9** [comment]
@jonscyr hi, this has already been fixed and released in the new version of Elastalert. 
ref: https://github.com/Yelp/elastalert/compare/v0.2.1...v0.2.2

**Comment by jonscyr** [comment]
yeah.. but it doesnt handle stopping the job everywhere. example `https://github.com/Yelp/elastalert/compare/v0.2.1...v0.2.2#diff-ffbe62c2238948ee1143861b0afeb3caR1088`

**Comment by abhishekjiitr** [comment]
The problem still exists in 0.2.4

**Comment by Shashankft9** [comment]
I see, in my testing I might not have covered all the test cases. Lets hope it gets merged. (please mention Qmando on your PR as well)

**Comment by Shashankft9** [comment]
@abhishekjiitr on another note were you able to initiate some activity on the prometheus plugin? I have been really looking forward to this, or please chime in if you have come up with some workaround?

**Comment by kylezs** [comment]
@Shashankft9 I'm using the `bitsensor/elastalert:3.0.0-beta.0` image, and was getting this issue.  A workaround: The deleted alerts stop after saving a new rule.

**Comment by Shashankft9** [comment]
I was just testing with slack and email, and got to face this issue. Th PR hasnt been merged yet :( 
@kylezs kind of hacky but works indeed, thanks.
